File: fibo.cpp
#include <vector>

void fibonacci(unsigned int x, unsigned long long int y, unsigned long long int z) {
	std::vector<unsigned long long int> fibo;
	fibo.push_back(y);
	fibo.push_back(z);
	printf("1: %llu\n2: %llu\n", fibo[0], fibo[1]);
	for (unsigned int i = 2; i < x; i++) {
		fibo.push_back(fibo[i-1] + fibo[i-2]);
		printf("%u: %llu\n", i+1, fibo[i]);
	}
}

int main(int argc, char* argv[]) {
	if (argc != 4) {
		printf("You have to provide 3 arguments - first number, secound number, how many numbers to calculate.\nExample:\t./fibo 0 1 100\n");
		return 1;
	}
	fibonacci(atoi(argv[3]), atoi(argv[1]), atoi(argv[2]));
	return 0;
}

File: fibo.rs
use std::env;
use std::process::exit;

fn fibo(x: usize, y: u128, z: u128) {
    let mut fibonacci: Vec<u128> = [y, z].to_vec();
    println!("1: {}\n2: {}", fibonacci[0], fibonacci[1]);
    for i in 2..x {
        fibonacci.push(fibonacci[i - 1] + fibonacci[i - 2]);
        println!("{}: {}", i + 1, fibonacci[i]);
    }
}

fn main() {
    let argv: Vec<String> = env::args().collect();
    if argv.len() != 4 {
        println!("You have to provide 3 arguments - first number, secound number, how many numbers to calculate.\nExample:\t./fibo 0 1 10\n");
        exit(1);
    }
    fibo(
        argv[3].parse::<usize>().unwrap(),
        argv[1].parse::<u128>().unwrap(),
        argv[2].parse::<u128>().unwrap(),
    );
    exit(0);
}
